src/jsCodegen/emitStep.ts

```diff
@@ -91,10 +93,13 @@ function emitParentAxis(
 	test: string,
 	predicates: string,
 	nestLevel: number,
-	nestedCode: string
+	nestedCode: string,
+	bucket: string | null
 ): PartialCompilationResult {
 	const contextNodeCode = `
-	const contextItem${nestLevel} = domFacade.getParentNode(contextItem${nestLevel - 1});
+	const contextItem${nestLevel} = domFacade.getParentNode(contextItem${nestLevel - 1} ${
+		bucket ? `, "${bucket}"` : ''
```

Comment:
why not pass `null` in here?

Reply:
That would also be possible but now it doesn't pass anything if bucket is null which makes the generated code somewhat cleaner and easier to read.

Reply:
aaaah, makes sense. awesome